github/packages.go

```diff
@@ -141,3 +212,106 @@ type PackageContainerMetadata struct {
 	return Stringify(r)
 }
+
+// PackageVersionBody represents the body field of a package version
+type PackageVersionBody struct {
+	Repo *Repository             `json:"repository,omitempty"`
+	Info *PackageVersionBodyInfo `json:"info,omitempty"`
+}
+
+func (b PackageVersionBody) String() string {
+	return Stringify(b)
+}
+
+// PackageVersionBodyInfo represents the info field of a PackageVersionBody
+type PackageVersionBodyInfo struct {
+	Type       *string `json:"type,omitempty"`
+	OID        *string `json:"oid,omitempty"`
+	Mode       *int64  `json:"mode,omitempty"`
+	Name       *string `json:"name,omitempty"`
+	Path       *string `json:"path,omitempty"`
+	Size       *int64  `json:"size,omitempty"`
+	Collection *bool   `json:"collection,omitempty"`
+}
+
+func (bi PackageVersionBodyInfo) String() string {
+	return Stringify(bi)
+}
+
+// PackageEventContainerMetadata represents container metadata for docker container packages. as part of a webhook event
+// See also PackageContainerMetadata
+type PackageEventContainerMetadata struct {
+	Labels   map[string]any                    `json:"labels,omitempty"`
+	Manifest map[string]any                    `json:"manifest,omitempty"`
+	Tag      *PackageEventContainerMetadataTag `json:"tag,omitempty"`
+}
+
+func (m PackageEventContainerMetadata) String() string {
+	return Stringify(m)
+}
+
+// PackageEventContainerMetadataTag represents a tag of a GitHub container package
+type PackageEventContainerMetadataTag struct {
+	Name   *string `json:"name,omitempty"`
+	Digest *string `json:"digest,omitempty"`
+}
+
+func (mt PackageEventContainerMetadataTag) String() string {
+	return Stringify(mt)
+}
+
+// PackageNugetMetadata represents nuget metadata for a GitHub package
+type PackageNugetMetadata struct {
+	ID    json.RawMessage `json:"id,omitempty"` // Can either be a int64 or string
+	Name  *string         `json:"name,omitempty"`
+	Value json.RawMessage `json:"value,omitempty"` // Can either be a bool, string, integer or object
+}
+
+func (nm PackageNugetMetadata) String() string {
+	return Stringify(nm)
+}
+
+// PackageNPMMetadata represents NPM metadata for a GitHub package
+type PackageNPMMetadata struct {
+	Name                 *string            `json:"name,omitempty"`
+	Version              *string            `json:"version,omitempty"`
+	NPMUser              *string            `json:"npm_user,omitempty"`
+	Author               *map[string]string `json:"author,omitempty"`
+	Bugs                 *map[string]string `json:"bugs,omitempty"`
+	Dependencies         *map[string]string `json:"dependencies,omitempty"`
+	DevDependencies      *map[string]string `json:"dev_dependencies,omitempty"`
+	PeerDependencies     *map[string]string `json:"peer_dependencies,omitempty"`
+	OptionalDependencies *map[string]string `json:"optional_dependencies,omitempty"`
+	Description          *string            `json:"description,omitempty"`
+	Dist                 *map[string]string `json:"dist,omitempty"`
```

Comment:
All of the `*map[string]string` should be `map[string]string` as maps are already reference types.

Same below.
I'm going to stop my review and let you address the issues found thus far.